contracts/SignatureVerifirer.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+
+abstract contract SignatureVerifier is EIP712 {
+
+  using ECDSA for bytes32;
+
+  struct BuyAuthorization {
+    address buyer;
+    uint nftId;    
+  }
+
+  bytes32 private constant BUY_AUTHORIZATION_TYPEHASH = keccak256("BuyAuthorization(address buyer, uint nftId)");
```

Comment:
I believe the standard says no space after the comma.

    BuyAuthorization(address buyer,uint nftId)